tests/python/relay/test_pass_fuse_ops.py

```diff
@@ -621,6 +621,81 @@ def expected():
     after = run_opt_pass(expected(), transform.InferType())
     assert tvm.ir.structural_equal(zz, after)
 
+
+def test_fuse_take():
```

Comment:
I may miss something, but could you elaborate a bit why fusion is related to this change?

Reply:
Sure. It may cause problems when fusing i32 ops into i64 ops. A simplified example:

```
i64_2 = tvm.tir.const(2, "int64")
i64_4 = tvm.tir.const(4, "int64")
i32_2 = tvm.tir.const(2, "int32")
a = te.placeholder((i64_4, i64_4), "a")
b = te.compute((i64_2, i64_2), lambda i, j: a[i + 2, j + 2])
c = te.compute((i32_2, i32_2), lambda i, j: b[i, j] + 1)
```

When fusing `b` into `c`, the i64 variables in `b` are replaced by i32 expressions in `c`, which does not make sense.